### DISCUSSION TITLE
How to integrate with sqlx

### DISCUSSION THREAD

**Discussion by mikebelanger** [ntex-rs/ntex#262] [Q&A]
Hi guys,

I couldn't find an example of how to integrate with [sqlx](https://github.com/launchbadge/sqlx), so I tried cobbling one together.  But I'm unsure this makes sense.   Specifically, how different ntex requests can use the same sqlx connection pool.

Similar to the r2d2 integration article [here](https://ntex.rs/docs/databases), I tried passing sqlx's connection pool as a parameter into an ntex handler function.  But I had to move it, then deref the address (is that even what I'm doing?) . 

Sorry for my newbish question, but I have a feeling this isn't the best way, despite it working on my local machine.  Here's the code:

https://github.com/mikebelanger/ntex_sqlx_experiment/blob/main/src/main.rs

Any feedback appreciated, thanks!

**Comment by fafhrd91** [comment]
you should use official example for sqlx + tokio (Axum), there should no difference

**Comment by shivshankardayal** [comment]
Did you succeed or you still need help? I have successfully used SQLx with ntex. You can take a look at https://github.com/Nalanda-Labs/N-FERP/ It used 0.7 of ntex which I am upgrading to 1.0

**Reply by mikebelanger** [reply]
Thanks @shivshankardayal .  It wasn't really a question of succeeding, as my code technically does already.  It was more a question of is my code the "right way" of doing it.

I haven't really touched on this recently as I've decided to use something other than ntex.
